### PR TITLE
move the handlers, servers, etc needed for a kubeapiserver to the kubeapiserver

### PR DESCRIPTION
**Pull request by deads2k**
There are handlers that only the kubeapiserver will need after we split.  The last commit here moves most of them to the point of use to make the pulls smaller.

### CODE REVIEW

File: pkg/client/genericinformers/interface.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/informers"
+)
+
+type GenericResourceInformer interface {
+	ForResource(resource schema.GroupVersionResource) (informers.GenericInformer, error)
+	Start(stopCh <-chan struct{})
+}
+
+// genericInternalResourceInformerFunc will return an internal informer for any resource matching
```
Comment (mfojtik) [inline pkg/client/genericinformers/interface.go]: nit: fix godoc

File: pkg/cmd/openshift-apiserver/openshiftapiserver/config.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+	authorizationOptions.RemoteKubeConfigFile = openshiftAPIServerConfig.MasterClients.OpenShiftLoopbackKubeConfig
+	if err := authorizationOptions.ApplyTo(&genericConfig.Authorization); err != nil {
+		return nil, err
+	}
+
+	informers, err := NewInformers(kubeInformers, kubeClientConfig, genericConfig.LoopbackClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	if err := informers.GetOpenshiftUserInformers().User().V1().Groups().Informer().AddIndexers(cache.Indexers{
```
Comment (mfojtik) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/config.go]: @deads2k i wonder if this should be moved to the controller that use this index?

File: pkg/cmd/openshift-apiserver/openshiftapiserver/configprocessing/audit.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	auditpolicy "k8s.io/apiserver/pkg/audit/policy"
+	auditlog "k8s.io/apiserver/plugin/pkg/audit/log"
+	auditwebhook "k8s.io/apiserver/plugin/pkg/audit/webhook"
+	pluginwebhook "k8s.io/apiserver/plugin/pkg/audit/webhook"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/apis/config"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+)
+
+func GetAuditConfig(auditConfig configapi.AuditConfig) (audit.Backend, auditpolicy.Checker, error) {
```
Comment (mfojtik) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/configprocessing/audit.go]: /cc @soltysh

File: pkg/cmd/openshift-apiserver/openshiftapiserver/images.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	kapi "k8s.io/kubernetes/pkg/apis/core"
+	coreinternalinformer "k8s.io/kubernetes/pkg/client/informers/informers_generated/internalversion/core/internalversion"
```
Comment (mfojtik) [inline pkg/cmd/openshift-apiserver/openshiftapiserver/images.go]: seems like low hanging fruit to switch to external?

**Comment by mfojtik** [comment]
@deads2k this looks awesome, tagged @soltysh to review the audit parts, otherwise LGTM

File: pkg/cmd/openshift-apiserver/openshiftapiserver/config.go (hunk 2 above)
Comment (deads2k) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/config.go]: > @deads2k i wonder if this should be moved to the controller that use this index?

it's in the server as I recall.  Perhaps, but not here.

File: pkg/cmd/openshift-apiserver/openshiftapiserver/images.go (hunk 4 above)
Comment (deads2k) [reply pkg/cmd/openshift-apiserver/openshiftapiserver/images.go]: > seems like low hanging fruit to switch to external?

probably, but not here.

**Pull request merged**